### PR TITLE
Add integrator contract

### PR DESCRIPTION
**Pull request by hugoroussel**

### CODE REVIEW

File: src/FantasyCardsProxy.sol [hunk 1]
```diff
@@ -0,0 +1,355 @@
...
+     * @param from address of the sender
+     * @param to address of the recipient
+     * @param tokenId tokenId
+     */
+    function safeTransferFrom(
+        address from,
+        address to,
+        uint256 tokenId
+    ) external override whenNotPaused approvedIntegrator {
+        return transferFrom(from, to, tokenId);
```
Comment (bane-the-dev) [inline src/FantasyCardsProxy.sol]:
```suggestion
       safeTransferFrom(from, to, tokenId, "");
```

File: src/FantasyCardsProxy.sol [hunk 2]
```diff
@@ -0,0 +1,355 @@
...
+     *
+     * WARNING: This function assumes that `owner` is the actual owner of `tokenId` and does not verify this
+     * assumption.
+     */
+    function _checkAuthorized(address owner, address spender, uint256 tokenId) internal view virtual {
+        if (!_isAuthorized(owner, spender, tokenId)) {
+            if (owner == address(0)) {
+                revert("Token does not exist");
+            } else {
+                // revert ERC721InsufficientApproval(spender, tokenId);
```
Comment (bane-the-dev) [inline src/FantasyCardsProxy.sol]:
```suggestion

```

Comment (bane-the-dev) [reply src/FantasyCardsProxy.sol]: Why do we use custom reverts here and not the standard ERC721?

File: src/interfaces/IFantasyCardsProxy.sol [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+     * @dev Emitted when `owner` enables `approved` to manage the `tokenId` token.
+     */
+    event Approval(address indexed owner, address indexed approved, uint256 indexed tokenId);
+
+    /**
+     * @dev Emitted when `owner` enables or disables (`approved`) `operator` to manage all of its assets.
+     */
+    event ApprovalForAll(address indexed owner, address indexed operator, bool approved);
+
+    event NewExecutionDelegate(address _executionDelegate);
```
Comment (bane-the-dev) [inline src/interfaces/IFantasyCardsProxy.sol:15]: group events together

File: src/FantasyCardsProxy.sol (hunk 2 above)
Comment (hugoroussel) [reply src/FantasyCardsProxy.sol]: dead code

**Pull request merged**